aliasing/api/context.py

```diff
@@ -243,13 +246,56 @@ def display_name(self):
         """
         return self._display_name
 
+    @property
+    def roles(self):
+        """
+        The user's roles. When used in a DM, it is always empty.
+        """
+        return self._roles
```

Comment:
```suggestion
    def roles(self):
        """
        The user's roles. When used in a DM, it is always empty.
        """
        return list(self._roles)
```

Reply:
Even without the generator, this would probably be a beneficial change, as we don't want the list inside `ctx.author` to be modified.